.github/workflows/test-move-prover.yml

```diff
@@ -1,38 +1,78 @@
 name: Test_Move_Prover
 
 on:
+  pull_request:
+  # types: [opened, reopened, synchronize]
   repository_dispatch:
     types: [new_PR]
 
 jobs:
   install_and_test_Move_Prover:
 
-    runs-on: ubuntu-latest
-
+    runs-on: ubuntu-18.04
+  
     steps:
+    - name: Checkout libra
+      uses: actions/checkout@v2
+      with:
+        path: "libra"
+        repository: ${{ github.event.client_payload.owner }}/${{ github.event.client_payload.repo }} # repo of the forked libra/libra
+        ref: ${{ github.event.client_payload.pull_ref }} # commit that triggered the PR
+    - name: Detect the changes in move-prover
+      working-directory: ./libra
+      run: |
+        git fetch origin
+        git diff --name-only origin/master
+        git diff --name-only origin/master | grep "language/move-prover"
+
     - name: Install Z3
       uses: pavpanchekha/setup-z3@v1.2
+      with:
+        version: "4.8.7"
     - name: Install .NET Core SDK and Boogie
       run: |
+        wget -q https://packages.microsoft.com/config/ubuntu/18.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+        sudo dpkg -i packages-microsoft-prod.deb
         sudo apt-get update
         sudo apt-get install apt-transport-https
         sudo apt-get update
         sudo apt-get install dotnet-sdk-3.1
-        dotnet tool install --global Boogie
+        dotnet tool install --global Boogie --version 2.5.7
+
     - name: Export the environment variables
       run: |
         echo "::set-env name=BOOGIE_EXE::/home/runner/.dotnet/tools/boogie"
         echo "::set-env name=Z3_EXE::`which z3`"
-    - name: Checkout libra
-      uses: actions/checkout@v2
-      with:
-        path: "libra"
-        repository: ${{ github.event.client_payload.owner }}/${{ github.event.client_payload.repo }} # repo of the forked libra/libra
-        ref: ${{ github.event.client_payload.pull_ref }} # commit that triggered the PR
-    - name: Build move-prover
-      working-directory: ./libra/language/move-prover/bytecode-to-boogie
-      run: cargo build
     - name: Test move-prover
       working-directory: ./libra/language/move-prover/bytecode-to-boogie
-      timeout-minutes: 10 # normally, expected to be done within 1 min.
-      run: cargo test
+      timeout-minutes: 30 # normally, expected to be done within 10 mins.
+      run: |
+        set -o pipefail
+        cargo test 2>&1 | tee cargo_test.log
+    - name: Comment the error message if the test failed
+      if: failure()
+      uses: actions/github-script@0.4.0
+      with:
+      #  github-token: ${{secrets.GITHUB_TOKEN}}
+        github-token: ${{secrets.MIRAI_BOT}}
```

Comment:
this will post a comment using the MIRAI bot, which is fine by me but you might want to personalize the message to really emphasis that it is not a MIRAI finding